src/pages/CarsList/CarsList.tsx

```diff
@@ -1,66 +1,62 @@
 import React, { useEffect, useState } from "react";
+import { Button, ButtonGroup, Container } from "react-bootstrap";
 import { useDispatch, useSelector } from "react-redux";
-import { CarDetails } from "../../models/CarDetails";
-import { RootState } from "../../redux/configureStore";
-import { getCars } from "../../redux/store/cars/actions";
+import { useLocation, useNavigate } from "react-router-dom";
 import CarCard from "../../components/CarCard/CarCard";
+import { carTypeList } from "../../models/CarType";
+import { getCars } from "../../redux/store/cars/actions";
 import "./CarsList.scss";
-import { Button, ButtonGroup, Container } from "react-bootstrap";
-import { useLocation } from "react-router-dom";
-
 function CarsList() {
   const dispatch = useDispatch();
   const location = useLocation();
-  const [searchtext] = useState<string>(
-    location.state === null ? "" : (location.state as string)
-  );
-
+  const navigate = useNavigate();
+  const queryParams = new URLSearchParams(window.location.search);
```

Comment:
try using/modifying the `useQuery` hook mentioned here: https://v5.reactrouter.com/web/example/query-parameters